### ISSUE TITLE
How to set entity from widget programmatically ?

### ISSUE THREAD

**Issue by lucasKapf** [thingsboard/thingsboard#5705]
Hello,

Does anyone knows how to set entity from widget programmatically ?
I am creating a custom entities table widget.
I would like that this widget does the same thing as the "Set entity from Widget" but programmatically.
Is it possible ?

Regards,

Lucas Arsicaud

**Comment by JacksonBowe** [comment]
Yeah, you need to pass the widget in through the `params` argument of the `widgetContext.stateController`. See below:
```
function updateDashboardState(stateId, deviceLabel) {
    var params = {
        entityId: entityId,
        entityName: entityName,
        entityLabel: deviceLabel
    };
    widgetContext.stateController.openState(stateId,
        params, false);
    // ^ Could also use updateState method if you want to update current state instead of open new
}
```

**Comment by lucasKapf** [comment]
Hello @JacksonBowe, thank you for your answer. I used the updateState method, but it doesn't set my "selectedEquipement" alias. I don't know where to precise the state entity parameter name that need to be set in my code.
I want to reproduce the same thing as the thing done in the original widget (see image below).
![setEntityWidget](https://user-images.githubusercontent.com/95343678/145542471-6097bb81-555a-43b6-b1ac-c159525cb1e0.png)

**Comment by JacksonBowe** [comment]
Under the type of action you have selected `Update current dashboard state`. Select the `Custom action` option. Paste my code there. You'll need to make a little modification maybe. Check out the thingsboard youtube channel. There's a few videos covering what you want to do.

**Comment by ChantsovaEkaterina** [comment]
Hello, @lucasKapf 
The code sample to update the dashboard state programmatically using state entity parameter is below:
```
var stateParams = widgetContext.stateController.getStateParams();
stateParams.targetStateParameter = {
    entityId: entityId,
    entityName: entityName,
    entityLabel: entityLabel
};
widgetContext.stateController.updateState('targetStatedId', stateParams, false);
```

**Comment by lucasKapf** [comment]
Hello @ChantsovaEkaterina, thank you very much for your answer, thank to you I was able to do what I wanted

**Comment by DurandA** [comment]
On more recent version of ThingsBoard (I am using 3.5.1PE), the API is different as the target entity is nested:
```javascript
let params = {
    myTargetEntityName: {
        entityId: entityId,
        entityName: entityName,
        entityLabel: entityLabel
    },
    targetEntityParamName: 'myTargetEntityName'
};
self.ctx.stateController.updateState(null, params, false);
```

**Comment by SindreLindahl** [comment]
This isn't working for me. I can update state and navigate to other dashboard. But the entity parameter name (and accordingly the entity) isn't applied in the other dashboard.

**Comment by nicovon24** [comment]
Hello guys, am i able also to select multiple entities, instead of an object been an array of values? Thanks!
